File: Merge_Sorted_Array.py
# 88. Merge Sorted Array (easy)
# my original solution

class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        for i in range(len(nums1) - 1, -1, -1):
            if 0!=nums1[i] or i == (m-1):
                break
            else:
                del nums1[i]

        nums1 += nums2

        nums1.sort()


# better alternative
'''
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        nums1[:] = sorted(nums1[:m]+nums2)
'''

File: Merge_Sorted_Array.js
// 88. Merge Sorted Array (easy)
/**
 * @param {number[]} nums1
 * @param {number} m
 * @param {number[]} nums2
 * @param {number} n
 * @return {void} Do not return anything, modify nums1 in-place instead.
 */
var merge = function(nums1, m, nums2, n) {
    for (let i = nums1.length-1; i>=0; i--) {
        if (nums1[i] !== 0 || i === m-1) {
            break;
        } else {
            nums1.pop()
        }
    }
    nums1.push(...nums2);

    nums1.sort((a, b) => a-b);
};

// alternative solution
var merge = function(nums1, m, nums2, n) {
    nums1.length = m;
    nums1.push(...nums2);

    nums1.sort((a, b) => a-b);
};
